File: src/test/java/net/haesleinhuepf/clij/ops/examples/OrthogonalMaximumProjections.java

package net.haesleinhuepf.clij.ops.examples;

import java.io.IOException;

import org.junit.Test;

import net.haesleinhuepf.clij.ops.CLIJ_close.CLIJ_close;
import net.haesleinhuepf.clij.ops.CLIJ_maximumXYZProjection.CLIJ_maximumXYZProjection;
import net.haesleinhuepf.clij.ops.CLIJ_pull.CLIJ_pull;
import net.haesleinhuepf.clij.ops.CLIJ_push.CLIJ_push;
import net.imagej.ImageJ;

public class OrthogonalMaximumProjections {

	@Test
	public void run() throws IOException {
		int angle_step = 10;
		ImageJ ij = new ImageJ();
		Object input = ij.io().open(getClass().getResource("/samples/t1-head.tif").getPath());
		ij.ui().show("input", input);
		// push image to GPU
		Object inputGPU = ij.op().run(CLIJ_push.class, input);

		float downScalingFactorInXY = 0.666f; // because the image has slice
																					// distance 1.5
		float downScalingFactorInZ = 1;

		Object downscaled = ij.op().run(
			net.haesleinhuepf.clij.ops.transform.downsample.CLIJ_downsample.class,
			inputGPU, downScalingFactorInXY, downScalingFactorInXY,
			downScalingFactorInZ);

		Object maximumProjectionX = ij.op().run(CLIJ_maximumXYZProjection.class,
			downscaled, 2, 1, 0);
		Object maximumProjectionY = ij.op().run(CLIJ_maximumXYZProjection.class,
			downscaled, 2, 0, 1);
		Object maximumProjectionZ = ij.op().run(CLIJ_maximumXYZProjection.class,
			downscaled, 0, 1, 2);

		// show results
		Object resultX = ij.op().run(CLIJ_pull.class, maximumProjectionX);
		Object resultY = ij.op().run(CLIJ_pull.class, maximumProjectionY);
		Object resultZ = ij.op().run(CLIJ_pull.class, maximumProjectionZ);
		ij.ui().show("Maximum projection in X", resultX);
		ij.ui().show("Maximum projection in Y", resultY);
		ij.ui().show("Maximum projection in Z", resultZ);

		// cleanup
		ij.op().run(CLIJ_close.class, inputGPU);
		ij.op().run(CLIJ_close.class, downscaled);
		ij.op().run(CLIJ_close.class, maximumProjectionX);
		ij.op().run(CLIJ_close.class, maximumProjectionY);
		ij.op().run(CLIJ_close.class, maximumProjectionZ);
		ij.op().run(CLIJ_close.class);

	}

	public static void main(String... args) throws IOException {
		OrthogonalMaximumProjections task = new OrthogonalMaximumProjections();
		task.run();
	}
}

File: src/test/resources/jython/orthogonalMaximumProjections.py
# CLIJ example ImageJ Ops jython: orthogonalMaximumProjections.py
#
# This script shows how maximum-X, -Y and -Z projections can be created using the GPU.
#
# Author: Robert Haase, rhaase@mpi-cbg.de
# Author: Deborah Schmidt, frauzufall@mpi-cbg.de
# May 2019
# ---------------------------------------------

#@ IOService io
#@ UIService ui
#@ OpService ops


import os
import inspect
def getResource(file):
    return os.path.dirname(os.path.abspath(inspect.getsourcefile(lambda:0))) + "/" + file


input = io.open(getResource("../samples/t1-head.tif"))
ui.show("input", input)
# push image to GPU
inputGPU = ops.run("CLIJ_push", input)

downScalingFactorInXY = 0.666; # because the image has slice distance 1.5
downScalingFactorInZ = 1;

downscaled = ops.run("CLIJ_downsample", inputGPU, downScalingFactorInXY, downScalingFactorInXY, downScalingFactorInZ)

maximumProjectionX = ops.run("CLIJ_maximumXYZProjection", downscaled, 2, 1, 0)
maximumProjectionY = ops.run("CLIJ_maximumXYZProjection", downscaled, 2, 0, 1)
maximumProjectionZ = ops.run("CLIJ_maximumXYZProjection", downscaled, 0, 1, 2)

# show results
resultX = ops.run("CLIJ_pull", maximumProjectionX)
resultY = ops.run("CLIJ_pull", maximumProjectionY)
resultZ = ops.run("CLIJ_pull", maximumProjectionZ)
ui.show("Maximum projection in X", resultX)
ui.show("Maximum projection in Y", resultY)
ui.show("Maximum projection in Z", resultZ)

#cleanup
ops.run("CLIJ_close", inputGPU)
ops.run("CLIJ_close", downscaled)
ops.run("CLIJ_close", maximumProjectionX)
ops.run("CLIJ_close", maximumProjectionY)
ops.run("CLIJ_close", maximumProjectionZ)
